### PR TITLE
feat(ng-uirouter-layout): add support for componentProvider

### PR DESCRIPTION
**Pull request by marie-j**
Ui Router allows to pass `componentProvider` for conditional component rendering. replicate that option for modal layouts

### CODE REVIEW

File: packages/components/ng-uirouter-layout/src/index.js [hunk 1]
```diff
@@ -69,10 +70,19 @@ angular
         }
 
         if (get(state, 'layout.name') === 'modal') {
+          let componentName = get(state, state.views.modal ? 'views.modal.component' : 'component');
```
Comment (frenautvh) [inline packages/components/ng-uirouter-layout/src/index.js]: get(state, "views.modal.component", "component") ?

if you're calling state.views.modal explicitly you loose the benefits of using lodash "get"

Comment (marie-j) [reply packages/components/ng-uirouter-layout/src/index.js]: Indeed

**Comment by marie-j** [comment]
> Hello,
> 
> Maybe we can provide a description of how to use it here:
> 
> https://github.com/ovh-ux/manager/tree/master/packages/components/ng-uirouter-layout
> 
> See ya

@jleveugle Sure, I will take care of this in another PR if that's ok with you ?

**Comment by antleblanc** [comment]
It probably make more sense to review and merge a pending pull request that is actually renaming this module.

ref: ovh-ux/manager#608

**Comment by marie-j** [comment]
> It probably make more sense to review and merge a pending pull request that is actually renaming this module.
> 
> ref: #608

@antleblanc The name of the module is not impacting this PR so I'm having trouble seeing why it makes more sense

**Comment by antleblanc** [comment]
Both will ended up with a conflict if we use the merge button.

#608 was already rebased several times that's why I have left a comment in order to refer to this pull request.

**Pull request merged**